### PR TITLE
expression, executor, plan: rewrite builtin function exp

### PR DESCRIPTION
**Pull request by stdupp**

### CODE REVIEW

**Comment by zz-jason** [comment]
LGTM

**Comment by zz-jason** [comment]
@XuHuaiyu PTAL

File: expression/builtin_math_test.go [hunk 1]
```diff
@@ -122,34 +121,47 @@ func (s *testEvaluatorSuite) TestCeil(c *C) {
...
+		{int64(1), 2.718281828459045, false, false, ""},
+		{float64(1.23), 3.4212295362896734, false, false, ""},
+		{float64(-1.23), 0.2922925776808594, false, false, ""},
+		{float64(0), 1, false, false, ""},
+		{"0", 1, false, false, ""},
+		{"tidb", 0, false, true, ""},
+		{float64(100000), 0, false, true, "[types:1690]DOUBLE value is out of range in 'exp(100000)'"},
 	}
-	for _, t := range testcases {
-		if runtime.GOARCH == "ppc64le" && t.num == int64(1) {
```
Comment (stdupp) [inline expression/builtin_math_test.go]: Ok.. I will add it later.

**Comment by shenli** [comment]
LGTM

**Pull request merged**